pkg/controller/node/node_controller.go

```diff
@@ -197,23 +200,28 @@ func inSlice(str string, s []string) bool {
 	return false
 }
 
-func filterPortsByNodeAddresses(nsxClients *NsxClients, ports *[]nsxtmgr.LogicalPort, nodeAddresses []string) ([]*nsxtmgr.LogicalPort, error) {
-	var filteredPorts []*nsxtmgr.LogicalPort
+func filterPortsByNodeAddresses(nsxClients *NsxClients, ports *[]model.SegmentPort, nodeAddresses []string) ([]*model.SegmentPort, error) {
+	var filteredPorts []*model.SegmentPort
 	log.Info(fmt.Sprintf("Found %d ports for node, checking addresses %v", len(*ports), nodeAddresses))
-	nsxClient := nsxClients.ManagerClient
+	connector := nsxClients.PolicyConnector
+	segPortStateClient := segPortSDK.NewStateClient(connector)
 	for _, port := range *ports {
-		logicalPort, _, err := nsxClient.LogicalSwitchingApi.GetLogicalPortState(nsxClient.Context, port.Id)
+		segId, err := operatortypes.ExtractSegmentIdFromPath(*port.ParentPath)
 		if err != nil {
 			return filteredPorts, err
 		}
-		if len(logicalPort.RealizedBindings) == 0 {
+		portState, err := segPortStateClient.Get(segId, *port.Id, nil, nil)
+		if err != nil {
+			return filteredPorts, err
+		}
+		if len(portState.RealizedBindings) == 0 {
 			continue
 		}
 		var collectedAddresses []string
-		for _, realizedBinding := range logicalPort.RealizedBindings {
-			address := realizedBinding.Binding.IpAddress
+		for _, realizedBinding := range portState.RealizedBindings {
+			address := *realizedBinding.Binding.IpAddress
 			if inSlice(address, nodeAddresses) && !inSlice(address, collectedAddresses) {
-				log.Info(fmt.Sprintf("Node address %s matches port %s", address, port.Id))
+				log.Info(fmt.Sprintf("Node address %s matches port %s", address, *port.Id))
```

Comment:
nit: change this line to `node address %s matches segment port %s`.
In this way, when we will check logs for troubleshooting, we will know that the operator is executing the code for policy API.
Also, I think the linter suggests to start log lines with a lowercase letter, so it might be another good thing to do.

Reply:
done